src/etc.cr

```diff
@@ -0,0 +1,122 @@
+lib LibC
+  ifdef darwin
+    struct Passwd
+      name : Char*
+      passwd : Char*
+      uid : UInt32
+      gid : UInt32
+      change : TimeT
+      pwclass : Char*
+      gecos : Char*
+      dir : Char*
+      shell : Char*
+      expire : TimeT
+      fields : Int32
+    end
+  elsif linux
+    struct Passwd
+      name : Char*
+      passwd : Char*
+      uid : UInt32
+      gid : UInt32
+      gecos : Char*
+      dir : Char*
+      shell : Char*
+    end
+  end
+
+  fun getlogin : Char*
+  fun getpwnam(name : Char*) : Passwd*
+  fun getpwuid(uid : UInt32) : Passwd*
+  fun getuid : UInt32
+end
+
+# The `Etc` module provides access to currently logged in user and
+# information typically stored in files in the /etc directory on Unix
+# systems.
+#
+# ### Example
+#
+#     require "etc"
+#
+#     login = Etc.getlogin
+#     info = Etc.getpwnam(login)
+#     full_name = info.gecos.sub(/,.*/, "")
+#     puts "Hello #{full_name}!"
+#
+module Etc
+  struct Passwd
+    property name
+    property passwd
+    property uid
+    property gid
+    property gecos
+    property dir
+    property shell
+
+    def initialize(pwd : LibC::Passwd)
```

Comment:
Agreed. Naming is hard, but I pushed a new commit trying to fix that. (As a separate commit for clarity, will squash it later.)